app/src/main/java/org/mozilla/focus/utils/FirebaseHelper.java

```diff
@@ -219,6 +220,9 @@ protected Void doInBackground(Void... voids) {
                         settings.setNewsSource(source);
                         NewsSourceManager.getInstance().setNewsSource(source);
                     }
+
+                    final String url = AppConfigWrapper.getLifeFeedProviderUrl(applicationContext, settings.getNewsSource());
+                    ThreadUtils.postToMainThread(() -> NewsSourceManager.getInstance().setNewsSourceUrl(url));
```

Comment:
why post to main thread?

Reply:
setNewsSourceUrl() will trigger series actions related to UI update.
setNewsSourceUrl() -> Repository.nextSubscription() -> onDataChangedListener::onDataChanged() -> ContentViewModel.item.setValue()

Reply:
Please correct me if I'm wrong. 
Repository.nextSubscription() -> Could run in background thread
onDataChangedListener::onDataChanged() -> it's called in LiveData's onChanged so it's UI thread.
ContentViewModel.item.setValue() -> same as above